### PR TITLE
Bugfix a single na n value spoils the entire diagram

### LINKED ISSUE #162: a single NaN value spoils the entire diagram
and nothing is displayed

### PR DESCRIPTION
**Pull request by JoachimCoenen**
fixes: #162.

### CODE REVIEW

File: core/typ/range.cpp [hunk 1]
```diff
@@ -80,7 +72,7 @@ double Range::center() const
 Range Range::slice(int i, int n) const
 {
     if (!isValid())
-        THROW("BUG: Range::slice called for invalid range");
+            THROW("BUG: Range::slice called for invalid range");
```
Comment (jwuttke) [inline core/typ/range.cpp]: restore indentation, please

File: gui/view/plot_diagram.cpp [hunk 2]
```diff
@@ -70,10 +70,30 @@ void PlotDiagram::refresh()
...
-        graph_->setDataValueError(QVector<double>::fromStdVector(xs), QVector<double>::fromStdVector(ys), QVector<double>::fromStdVector(ysSigma));
-    else
-        graph_->setData(QVector<double>::fromStdVector(xs), QVector<double>::fromStdVector(ys));
+    if (ysSigma.size() > 0) {// has valueError
+        std::vector<double> xsSafe;
+        std::vector<double> ysSafe;
+        std::vector<double> ysSigmaSafe;
+        for (size_t i = 0; i < xs.size(); ++i)
+            if (   !qIsNaN(xs.at(i)) && qIsFinite(xs.at(i))
+                && !qIsNaN(ys.at(i)) && qIsFinite(ys.at(i))) {
```
Comment (jwuttke) [inline gui/view/plot_diagram.cpp]: bitte anders herum:

for (...) {
   if (anomaler Zustand)
      continue;
   normale Prozedur;
}

**Review by jwuttke — COMMENTED** [review]
see inlined notes

**Comment by JoachimCoenen** [comment]
done!

**Review by jwuttke — APPROVED** [review]

**Pull request merged**